src/testcases/ClassesTest.java

```diff
@@ -1,5 +1,454 @@
+     */
+    @Test
+    public void testCreateClass()
+    {
+        Classes classes = new Classes();
+        classes.createClass("E1");
+        assertTrue("Entities list size is 1", classes.getEntities().size() == 1);
+        assertTrue("Entities list contains E1", classes.getEntities().contains(new Entity("E1")));
+
+        assertFalse("createClass returned false after called with name of existing class", classes.createClass("E1"));
+
+        classes.createClass("E2");
+        classes.createClass("E3");
+        assertTrue("Entities list size is 3", classes.getEntities().size() == 3);
+        assertTrue("Entities list contains E2", classes.getEntities().contains(new Entity("E2")));
+        assertTrue("Entities list contains E3", classes.getEntities().contains(new Entity("E3")));
+    }
+
+    /** test the renameClass method
+     *
+     * ensure that classes can be renamed properly
+     */
+    @Test
+    public void testRenameClass()
+    {
+        Classes classes = new Classes();
+        classes.createClass("E1");
+
+        assertFalse("renameClass returned false after called with name of non-existing class", classes.renameClass("Non-exist", "e"));
+
+        classes.renameClass("E1", "E2");
+        assertTrue("Entities list contains E2", classes.getEntities().contains(new Entity("E2")));
+        assertFalse("Entities list does not contain E1", classes.getEntities().contains(new Entity("E1")));
+
+        classes.createClass("E1");
+        assertTrue("Entities list contains E1", classes.getEntities().contains(new Entity("E1")));
+
+        assertFalse("renameClass returned false after renaming to already existing class", classes.renameClass("E1", "E2"));
+
+        classes.createField("E1", "f");
+        classes.createMethod("E1", "m");
+
+        Entity e1_copy = classes.copyEntity("E1");
+        classes.renameClass("E1", "e");
+        Entity e = classes.copyEntity("e");
+
+        assertTrue("Class renamed to e", classes.getEntities().contains(e));
+        assertFalse("Class E1 no longer exists", classes.getEntities().contains(e1_copy));
+        assertTrue("Fields still in tact", e.getFields().equals(e1_copy.getFields()));
+        assertTrue("Methods still in tact", e.getMethods().equals(e1_copy.getMethods()));
+
+        /* Make sure class names were changed in existing relationships */
+        classes.createRelationship("r1", "e", "E2");
+        classes.renameClass("E2", "e2");
+
+        assertTrue("Class name (2) in relationship was updated", classes.searchRelationship("r1", "e", "e2"));
+        assertFalse("Old relationship no longer exists", classes.searchRelationship("r1", "e", "E2"));
+
+        classes.renameClass("e", "e1");
+        assertTrue("Class name (1) in relationship was updated", classes.searchRelationship("r1", "e1", "e2"));
+        assertFalse("Old relationship no longer exists", classes.searchRelationship("r1", "e", "e2"));
```

Comment:
The second class name should be a capital "E2", or else this assertFalse will fail

Reply:
I rename "E2" to "e2" earlier on line 230. I then renamed 'e' to 'e1', so now I am using assertFalse to check that the relationship r1 between e and e2 no longer exists, because it should now be between e1 and e2 after the renameClass calls.

Reply:
Ah you're right. I was looking at the other assertFalse above it that used "E2" before, and thought the bottom one should mirror it.